### PR TITLE
add(bot): ignore merge commits for coauthors

### PR DESCRIPTION
**Pull request by chdsbd**
Authors of merge commits probably shouldn't be considered coauthors unless they have made other commits to the pull request.

related #576

### CODE REVIEW

File: bot/kodiak/evaluation.py [hunk 1]
```diff
@@ -191,7 +192,16 @@ def get_merge_body(
...
         )
     if config.merge.message.include_coauthors:
-        coauthors += commit_authors
+        for commit in commits:
+            if (
+                # only use commits that have identified authors.
+                commit.author is None
+                or commit.author.user is None
+                # ignore merge commits. They will have more than one parent.
+                or commit.parents.totalCount > 1
```
Comment (chdsbd) [inline bot/kodiak/evaluation.py:201]: Beyond all the refactoring this is the big change.

File: bot/kodiak/queries/commits.py [hunk 2]
```diff
@@ -50,9 +55,4 @@ def get_commit_authors(*, pr: Dict[str, Any]) -> List[User]:
     nodes = pull_request.commitHistory.nodes
     if not nodes:
         return []
-    commit_authors = []
-    for node in nodes:
-        if node.commit.author is None or node.commit.author.user is None:
-            continue
-        commit_authors.append(node.commit.author.user)
-    return commit_authors
```
Comment (chdsbd) [inline bot/kodiak/queries/commits.py:58]: This logic now lives in `evaluation.py`

File: bot/tests/fixtures.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from typing import Optional
```
Comment (chdsbd) [inline bot/tests/fixtures.py:1]: I think we might want to move all the test stuff out of `kodiak/` eventually to make it easier to understanding what is for testing.

File: bot/kodiak/test_queries.py [hunk 4]
```diff
@@ -39,9 +39,11 @@
     StatusContext,
     StatusState,
     Subscription,
-    get_commit_authors,
+    get_commits,
 )
+from kodiak.queries.commits import CommitConnection, GitActor
 from kodiak.test_utils import wrap_future
+from tests.fixtures import create_commit
```
Comment (sbdchd) [inline bot/kodiak/test_queries.py:46]: I think it would be better if this was named spaced `kodiak.tests.fixtures`

**Review by sbdchd — APPROVED** [review]

**Pull request merged**